tests/test_template.py

```diff
@@ -70,56 +87,130 @@ def circuit(theta, phi):
                                             [ 3.81099656e-04, -4.79703149e-02,  5.78754312e-01,  0.00000000e+00, 3.38965962e-02,  0.00000000e+00]]
                                   ), delta=self.tol)
 
+class TestCVNeuralNet(BaseTest):
+    """Tests for the Interferometer from the pennylane.template module."""
+
+    num_subsystems = 4
+
     def setUp(self):
         super().setUp()
         np.random.seed(8)
         self.num_wires = 4
-
-    def test_cvqnn(self):
-
-        dev = qml.device('default.gaussian', wires=self.num_wires)
-
-        weights = [[
+        self.weights = [[
             np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #transmittivity angles
             np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #phase angles
             np.random.uniform(0.1, 0.7, self.num_wires), #squeezing amounts
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #squeezing angles
             np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #transmittivity angles
             np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #phase angles
             np.random.uniform(0.1, 2.0, self.num_wires), #displacement magnitudes
+            np.random.uniform(0.1, 2*np.pi, self.num_wires), #displacement angles
             np.random.uniform(0.1, 0.3, self.num_wires) #kerr parameters
         ] for l in range(2)]
 
+    def test_integration(self):
+        """integration test for the Interferomerter."""
+        dev = qml.device('default.gaussian', wires=self.num_subsystems)
+
+        # to test whether the correct circuit is produces we inspect the
+        # Device._op_queue by patching Device.execute()
+        test = self
+        test_weights = self.weights
+        def new_execute(self, queue, expectation):
+            test.assertAllEqual([type(op) for op in queue], [qml.Beamsplitter]*6+[qml.Squeezing]*4+[qml.Beamsplitter]*6+[qml.Displacement]*4+[qml.Kerr]*4+[qml.Beamsplitter]*6+[qml.Squeezing]*4+[qml.Beamsplitter]*6+[qml.Displacement]*4+[qml.Kerr]*4)
+
+            split_queue = np.split(queue, np.cumsum([6,4,6,4,4,6,4,6,4]))
+
+            test.assertAllEqual([op.parameters for op in split_queue[0]], [[theta_1, phi_1] for theta_1, phi_1 in zip(test_weights[0][0],test_weights[0][1])])
```

Comment:
This feels clunky. @josh146 any ideas how to streamline this test?

Reply:
I am not sure. I can of course generate the assert statements with a code similar to the code in `template.py` that I am testing, but that would largely defeat the purpose, no?

Reply:
I did make that part a little more elegant. Hope it is fine now.